blogService/src/main/java/blogService/service/UserService.java

```diff
@@ -36,12 +72,65 @@ public UserDto getUser(Long userId) {
 
     public String deleteUser(Long userId, String password) {
         Optional<User> optional = userRepository.findById(userId);
-        if (optional.isPresent()) {
-            if (password.hashCode() == userRepository.findById(userId).get().getPassHash()) {
-                userRepository.deleteById(userId);
-                return "The user has been deleted.";
-            }
+        if (optional.isPresent() && password.hashCode() == userRepository.findById(userId).get().getPassHash()) {
+            userRepository.deleteById(userId);
+            return "The user has been deleted.";
         }
         return "The user has not been deleted.";
     }
+
+    public void resetPassword(final String email) {
+        final User user = userRepository.findByEmail(email).orElseThrow(IllegalArgumentException::new);
+        final String temporaryPassword = RandomStringUtils.randomAlphanumeric(6);
+        user.setPassHash(temporaryPassword.hashCode());
+        userRepository.save(user);
+
+        sendMessage(temporaryPassword, email);
```

Comment:
byłoby ekstra, gdyby ta metoda zwracała boolean czy wiadomość została faktycznie wysłana (true - ok, false, coś poszło nie tak) w przyszłości pomoże to przy obsłudze błędów i pozwoli zwrócić wiadomość użytkownikowi na frontendzie

Reply:
a jeśli nie boolean to string, na tej samej zasadzie co w deleteUser :)

Reply:
zrobione (wersja String, dla spójności)